connector/logs_router.go

```diff
@@ -8,16 +8,16 @@ import (
 
 	"go.uber.org/multierr"
 
-	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/internal/fanoutconsumer"
+	"go.opentelemetry.io/collector/pipeline"
 )
 
 // LogsRouterAndConsumer feeds the first consumer.Logs in each of the specified pipelines.
 type LogsRouterAndConsumer interface {
 	consumer.Logs
-	Consumer(...component.ID) (consumer.Logs, error)
-	PipelineIDs() []component.ID
+	Consumer(...pipeline.ID) (consumer.Logs, error)
+	PipelineIDs() []pipeline.ID
```

Comment:
This (and the ones on metrics and traces router) are breaking changes, it seems hard to work around them so we can just add an API changelog entry